be/src/vec/exec/vjson_scanner.cpp

```diff
@@ -0,0 +1,517 @@
+    return Status::OK();
+}
+
+Status VJsonScanner::open_vjson_reader() {
+    if (_cur_vjson_reader != nullptr) {
+        _cur_vjson_reader.reset();
+    }
+    std::string json_root = "";
+    std::string jsonpath = "";
+    bool strip_outer_array = false;
+    bool num_as_string = false;
+    bool fuzzy_parse = false;
+
+    RETURN_IF_ERROR(JsonScanner::get_range_params(jsonpath, json_root, strip_outer_array,
+                                                  num_as_string, fuzzy_parse));
+    if (_read_json_by_line) {
+        _cur_vjson_reader.reset(new VJsonReader(_state, _counter, _profile, strip_outer_array,
+                                                num_as_string, fuzzy_parse, &_scanner_eof, nullptr,
+                                                _cur_line_reader));
+    } else {
+        _cur_vjson_reader.reset(new VJsonReader(_state, _counter, _profile, strip_outer_array,
+                                                num_as_string, fuzzy_parse, &_scanner_eof,
+                                                _cur_file_reader));
+    }
+
+    RETURN_IF_ERROR(_cur_vjson_reader->init(jsonpath, json_root));
+    return Status::OK();
+}
+
+VJsonReader::VJsonReader(RuntimeState* state, ScannerCounter* counter, RuntimeProfile* profile,
+                         bool strip_outer_array, bool num_as_string, bool fuzzy_parse,
+                         bool* scanner_eof, FileReader* file_reader, LineReader* line_reader)
+        : JsonReader(state, counter, profile, strip_outer_array, num_as_string, fuzzy_parse,
+                     scanner_eof, file_reader, line_reader),
+          _vhandle_json_callback(nullptr) {}
+
+VJsonReader::~VJsonReader() {}
+
+Status VJsonReader::init(const std::string& jsonpath, const std::string& json_root) {
+    // generate _parsed_jsonpaths and _parsed_json_root
+    RETURN_IF_ERROR(JsonReader::_parse_jsonpath_and_json_root(jsonpath, json_root));
+
+    //improve performance
+    if (_parsed_jsonpaths.empty()) { // input is a simple json-string
+        _vhandle_json_callback = &VJsonReader::_vhandle_simple_json;
+    } else { // input is a complex json-string and a json-path
+        if (_strip_outer_array) {
+            _vhandle_json_callback = &VJsonReader::_vhandle_flat_array_complex_json;
+        } else {
+            _vhandle_json_callback = &VJsonReader::_vhandle_nested_complex_json;
+        }
+    }
+
+    return Status::OK();
+}
+
+Status VJsonReader::read_json_column(std::vector<MutableColumnPtr>& columns,
+                                     const std::vector<SlotDescriptor*>& slot_descs,
+                                     bool* is_empty_row, bool* eof) {
+    return (this->*_vhandle_json_callback)(columns, slot_descs, is_empty_row, eof);
+}
+
+Status VJsonReader::_vhandle_simple_json(std::vector<MutableColumnPtr>& columns,
+                                         const std::vector<SlotDescriptor*>& slot_descs,
+                                         bool* is_empty_row, bool* eof) {
```

Comment:
does batch_size limit work here?

Reply:
yes, it read one batch_size every time.

Reply:
It seems not work good when _json_doc is array type

Reply:
we have finished the test, it will take effect.